kafka-rest/src/main/java/io/confluent/kafkarest/entities/v2/ProduceRequest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafkarest.entities.v2;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotEmpty;
+
+@AutoValue
+public abstract class ProduceRequest {
+
+  @NotEmpty
+  @JsonProperty("records")
+  public abstract ImmutableList<ProduceRecord> getRecords();
+
+  @JsonProperty("key_schema_id")
```

Comment:
are we going to loose the snake_case in the v3 api?

Reply:
Not really. The Confluent API Style Guide mandates lower underscore JSON attributes.